apps/namadillo/src/types.ts

```diff
@@ -202,3 +207,104 @@ export type AddressWithAssetAndAmountMap = Record<
   Address,
   AddressWithAssetAndAmount
 >;
+
+export const NamadaTransferProgressSteps = {
+  TransparentToShielded: [
+    "sign",
+    "zk-proof",
+    "tnam-to-masp",
+    "complete",
+  ] as const,
+  ShieldedToTransparent: ["sign", "masp-to-tnam", "complete"] as const,
+  ShieldedToShielded: ["sign", "masp-to-masp", "complete"] as const,
+  TransparentToTransparent: ["sign", "tnam-to-tnam", "complete"] as const,
+};
+
+export const IbcTransferProgressSteps = {
+  TransparentToIbc: ["sign", "ibc-withdraw", "complete"] as const,
+  IbcToShielded: ["sign", "zk-proof", "ibc-to-masp", "complete"] as const,
+  IbcToTransparent: ["sign", "ibc-to-namada", "complete"] as const,
+};
+
+export const TransferProgressSteps = {
+  ...NamadaTransferProgressSteps,
+  ...IbcTransferProgressSteps,
+};
+
+export const TransparentTransferTypes: Array<
+  keyof typeof TransferProgressSteps
+> = [
+  "ShieldedToTransparent",
+  "TransparentToIbc",
+  "TransparentToTransparent",
+  "IbcToTransparent",
+];
+
+export const IbcTransferTypes: Array<keyof typeof TransferProgressSteps> = [
+  "IbcToTransparent",
+  "TransparentToIbc",
+  "IbcToShielded",
+];
+
+export const TransferProgressStepsEntry = [
+  ...new Set(Object.values(TransferProgressSteps).flat()),
+] as const;
+
+export type ProgressStepsOptions = (typeof TransferProgressStepsEntry)[number];
+
+export type NamadaTransferTxKind = keyof typeof NamadaTransferProgressSteps;
+
+export type IbcTransferTxKind = keyof typeof IbcTransferProgressSteps;
+
+export type TransferTxKind = NamadaTransferTxKind & IbcTransferTxKind;
+
+export type NamadaTransferProgress = {
+  [P in NamadaTransferTxKind]: {
+    type: P;
+    progressStatus: (typeof NamadaTransferProgressSteps)[P][number];
+  };
+}[NamadaTransferTxKind];
+
+export type IbcTransferProgress = {
+  [P in IbcTransferTxKind]: {
+    type: P;
+    progressStatus: (typeof IbcTransferProgressSteps)[P][number];
+  };
+}[IbcTransferTxKind];
+
+export type TransferProgress = IbcTransferProgress | NamadaTransferProgress;
+
+export type TransferTransactionBase = TransferProgress & {
+  rpc: string;
+  hash?: string;
+  amount: BigNumber;
+  denom: string;
+  tokenSymbol: string;
+  chainId: string;
+  sourceAddress: string;
+  destinationAddress: string;
+  feePaid?: BigNumber;
+  tipPaid?: BigNumber;
+  resultTxHash?: string;
+  errorMessage?: string;
+  status: MutationStatus;
```

Comment:
I'm still not sure. I imagine situations where we can add a transaction to a queue to process the zk proof, so it will be in idle state, waiting to start to get processed. wdyt?

Reply:
That could be possible! I think leaving `idle` is fine.